crates/bevy_hierarchy/src/one_to_one/component.rs

```diff
@@ -0,0 +1,188 @@
+#[cfg(feature = "reflect")]
+use bevy_ecs::reflect::{
+    ReflectComponent, ReflectFromWorld, ReflectMapEntities, ReflectVisitEntities,
+    ReflectVisitEntitiesMut,
+};
+use bevy_ecs::{
+    component::{Component, ComponentId},
+    entity::{Entity, VisitEntities, VisitEntitiesMut},
+    event::Events,
+    traversal::Traversal,
+    world::{DeferredWorld, FromWorld, World},
+};
+use core::fmt::Debug;
+use core::marker::PhantomData;
+use core::ops::Deref;
+
+use super::OneToOneEvent;
+
+/// Represents one half of a one-to-one relationship between two [entities](Entity).
+///
+/// The type of relationship is denoted by the parameter `R`.
+#[derive(Component)]
+#[component(
+    on_insert = Self::associate,
+    on_replace = Self::disassociate,
+    on_remove = Self::disassociate
+)]
+#[cfg_attr(feature = "reflect", derive(bevy_reflect::Reflect))]
+#[cfg_attr(
+    feature = "reflect",
+    reflect(
+        Component,
+        MapEntities,
+        VisitEntities,
+        VisitEntitiesMut,
+        PartialEq,
+        Debug,
+        FromWorld
+    )
+)]
+pub struct OneToOne<R> {
```

Comment:
This is a totally new feature and not strictly required to update `Parent` and `Children`. _However_, I found this to be a much simpler component to design, and I think it helps explain how the One-to-Many components work, while providing a new tool for users.

Reply:
In the future, we can maybe add ManyToMany relationship, it probably can be useful overall, not sure of the use cases.

Reply:
After refactoring how these components work, I've added `ManyToMany`, with directed and undirected forms for `OneToOne` and `ManyToMany`. I haven't added an undirected form of `OneToMany` as I'm not entirely sure what that actually represents, and I'd rather not invent it for the sake of completeness.